src/test-utils/render.tsx

```diff
@@ -31,23 +31,34 @@ const WithProviders = ({ children }: { children?: ReactNode }) => {
       <Provider store={store}>
         <QueryClientProvider client={queryClient}>
           <HashRouter>
-            <Web3Provider>
-              <MockedProvider>
-                <BlockNumberProvider>
-                  <ThemeProvider>{children}</ThemeProvider>
-                </BlockNumberProvider>
-              </MockedProvider>
-            </Web3Provider>
+            {/*
+             * Web3Provider is mocked through setupTests.ts
+             * To test behavior that depends on Web3Provider, use jest.unmock('@web3-react/core')
+             */}
```

Comment:
+1 on the comment. My worry is that this may become confusing in the future when analyzed outside of the context of this PR, where I can see where the `<Web3Provider />` used to be. Someone reading this file in the future may not be aware that such provider is needed in first place for our application to work.

Some alternatives:
- Define an empty component `const Web3Provider = (children) => children` that is just an empty pass-through and comment that component instead. That way, it provides direct explanation why we're not using Web3Provider from `web3-react`, but this one instead.
- Mock `components/Web3Provider` in `setupTests` next to where we mock `web3React` import.

Reply:
I think both alternatives add extra and unnecessary complexity. I'd prefer to just leave it as this comment. This is the place where the providers are setup, so I think it's appropriate to just leave aa comment here.